pyiface/iface.py

```diff
@@ -192,11 +192,12 @@ def __newIfreqWithName(self):
 
     def __doIoctl(self, ifr, SIOC, mutate=True):
         try:
-            with socket.socket(socket.AF_INET, socket.SOCK_DGRAM, 0) as skt:
-                fcntl.fcntl(skt,
-                            fcntl.F_SETFD,
-                            fcntl.fcntl(skt, fcntl.F_GETFD) | fcntl.FD_CLOEXEC)
-                fcntl.ioctl(skt, SIOC, ifr, mutate)
+            skt = socket.socket(socket.AF_INET, socket.SOCK_DGRAM, 0)
+            fcntl.fcntl(skt,
+                        fcntl.F_SETFD,
+                        fcntl.fcntl(skt, fcntl.F_GETFD) | fcntl.FD_CLOEXEC)
+            fcntl.ioctl(skt, SIOC, ifr, mutate)
+            skt.close()
```

Comment:
What about putting skt.close() into the `finally` section?

Reply:
something like:
```python3
if skt:
    skt.close()
```